### ISSUE TITLE
Upgrade Graphiql to enable headers

### ISSUE THREAD

**Issue by timothyschu** [graphql-java-kickstart/graphql-spring-boot#441]
**Is your feature request related to a problem? Please describe.**
Currently, the GraphiQL SpringBoot starter does not support adding dynamic headers. 

**Describe the solution you'd like**
With the newest version of GraphiQL, a new tab was added which allows for dynamic headers. `graphql-spring-boot` should use that latest version. See the GraphiQL issue provided under additional context.

**Describe alternatives you've considered**
The alternative is to set the header in the application config, but this is a static header. Changing headers requires a restart.

**Additional context**
https://github.com/graphql/graphiql/issues/59

**Comment by kyle-wang-sd** [comment]
It's finally resoved by https://github.com/graphql/graphiql/pull/1543, this is good.
But the whole process to apply this feature really really sucks, it have been 5 years since we want this feature.

**Comment by oliemansm** [comment]
We welcome PRs if anybody has the time to implement this.

**Comment by oliemansm** [comment]
I've upgraded GraphiQL to 1.0.6. Will be available in release 8.1.0. To enable the headers you can add the following application property:

```yaml
graphiql:
  variables:
    headerEditorEnabled: true
```

Just like you can configure any of the [GraphiQL Props](https://github.com/graphql/graphiql/tree/main/packages/graphiql#props) under the `graphiql.variables` group.

**Comment by BlasiusSecundus** [comment]
Apart of the issue addressed in the PR above, I noticed several issues with the header configuration: 

- headers configured in `graphiql.headers` are not present in the "Request headers" tab, but are sent with the request (assuming the content of the tab was not edited)
- headers configured in `graphiql.props.variables.headers` are present in the "Request headers" tab, but are not sent with the request (unless a content change event is triggered)
- if both are configured the content of  `graphiql.headers` is submitted but the content of `graphiql.props.variables.headers` is displayed

@oliemansm Since this is an unreleased feature, I am not sure if a new issue should be raised or this one should be reopened.

**Comment by craigmit** [comment]
> I've upgraded GraphiQL to 1.0.6. Will be available in release 8.1.0. To enable the headers you can add the following application property:
> 
> ```yaml
> graphiql:
>   props:
>     variables:
>       headerEditorEnabled: true
> ```
> 
> Just like you can configure any of the [GraphiQL Props](https://github.com/graphql/graphiql/tree/main/packages/graphiql#props) under the `graphiql.props.variables` group.

Adding `headerEditorEnabled: true` give me:
```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'graphiQLController': Invocation of init method failed; nested exception is com.fasterxml.jackson.databind.JsonMappingException: java.lang.Boolean cannot be cast to java.lang.String (through reference chain: java.util.Properties["headerEditorEnabled"])
```

Changing it to `headerEditorEnabled: "true"` (Ie: Giving it a string, not a boolean) fixes the problem.

Version 11.0.0.

**Comment by craigmit** [comment]
> I've upgraded GraphiQL to 1.0.6. Will be available in release 8.1.0. To enable the headers you can add the following application property:
> 
> ```yaml
> graphiql:
>   props:
>     variables:
>       headerEditorEnabled: true
> ```
> 
> Just like you can configure any of the [GraphiQL Props](https://github.com/graphql/graphiql/tree/main/packages/graphiql#props) under the `graphiql.props.variables` group.

Note:  You'll get an error about trying to convert a boolean to a string if you're running an old version of com.fasterxml.jackson.core.
